drivers/storage/portworx/portworx.go

```diff
@@ -408,91 +412,107 @@ func setNodeSpecDefaults(toUpdate *corev1alpha1.StorageCluster) {
 	toUpdate.Spec.Nodes = updatedNodeSpecs
 }
 
-func setComponentDefaults(
+func setDefaultAutopilotProviders(
 	toUpdate *corev1alpha1.StorageCluster,
-	components manifest.Release,
-	isPortworxEnabled bool,
 ) {
-	// Use the lighthouse image from release manifest if the current image is not locked,
-	// else keep using the existing image. If the current image is empty then use the
-	// default image from manifest else a hardcoded one if absent in manifest.
-	if toUpdate.Spec.UserInterface != nil &&
-		toUpdate.Spec.UserInterface.Enabled {
-		toUpdate.Spec.UserInterface.Image = strings.TrimSpace(toUpdate.Spec.UserInterface.Image)
-		if len(components.Lighthouse) > 0 {
-			if !toUpdate.Spec.UserInterface.LockImage ||
-				len(toUpdate.Spec.UserInterface.Image) == 0 {
-				toUpdate.Spec.UserInterface.Image = components.Lighthouse
-			}
-		} else if len(toUpdate.Spec.UserInterface.Image) == 0 {
-			toUpdate.Spec.UserInterface.Image = defaultLighthouseImage
+	if toUpdate.Spec.Autopilot != nil && toUpdate.Spec.Autopilot.Enabled &&
+		len(toUpdate.Spec.Autopilot.Providers) == 0 {
+		toUpdate.Spec.Autopilot.Providers = []corev1alpha1.DataProviderSpec{
+			{
+				Name: "default",
+				Type: "prometheus",
+				Params: map[string]string{
+					"url": "http://prometheus:9090",
+				},
+			},
 		}
 	}
+}
 
-	// Use the autopilot image from release manifest if the current image is not locked,
-	// else keep using the existing image. If the current image is empty then use the
-	// default image from manifest else a hardcoded one if absent in manifest.
-	if toUpdate.Spec.Autopilot != nil &&
-		toUpdate.Spec.Autopilot.Enabled {
-		toUpdate.Spec.Autopilot.Image = strings.TrimSpace(toUpdate.Spec.Autopilot.Image)
-		if len(components.Autopilot) > 0 {
-			if !toUpdate.Spec.Autopilot.LockImage ||
-				len(toUpdate.Spec.Autopilot.Image) == 0 {
-				toUpdate.Spec.Autopilot.Image = components.Autopilot
-			}
-		} else if len(toUpdate.Spec.Autopilot.Image) == 0 {
-			toUpdate.Spec.Autopilot.Image = defaultAutopilotImage
-		}
-
-		if len(toUpdate.Spec.Autopilot.Providers) == 0 {
-			toUpdate.Spec.Autopilot.Providers = []corev1alpha1.DataProviderSpec{
-				{
-					Name: "default",
-					Type: "prometheus",
-					Params: map[string]string{
-						"url": "http://prometheus:9090",
-					},
-				},
-			}
+func removeDeprecatedFields(
+	cluster *corev1alpha1.StorageCluster,
+) {
+	// For new clusters with this version of the operator we don't
+	// want to reset the images if they are present
+	existingCluster := cluster.Spec.Version != ""
+	// If the image has been reset once, we should not do it again,
+	// as the user could have chosen to override the image manually.
+	// status.version is a newly introduced field and will be
+	// populated first time when this version of operator runs.
+	alreadyDone := cluster.Status.Version != ""
+
```

Comment:
Ideally we can return from here as we won't use lockImage for new clusters.
This will just ensure if a user sets the lockImage flag post installation, we go ahead and reset it to avoid confusion.